homeassistant/components/simplisafe/config_flow.py

```diff
@@ -27,40 +32,77 @@ def __init__(self):
         return SimpliSafeOptionsFlowHandler(config_entry)
 
-    async def async_step_import(self, import_config):
-        """Import a config entry from configuration.yaml."""
-        return await self.async_step_user(import_config)
+    async def _async_get_simplisafe_api(self, user_input):
+        """Attempt to log into SimpliSafe."""
+        client_id = await async_get_client_id(self.hass)
+        websession = aiohttp_client.async_get_clientsession(self.hass)
+
+        return await API.login_via_credentials(
+            user_input[CONF_USERNAME],
+            user_input[CONF_PASSWORD],
+            client_id=client_id,
+            session=websession,
+        )
+
+    async def async_step_mfa(self, user_input=None):
+        """Handle the start of the config flow."""
+        if user_input is None:
+            return self.async_show_form(step_id="mfa")
+
+        try:
+            simplisafe = await self._async_get_simplisafe_api(self._post_mfa_user_input)
+        except SimplipyError as err:
+            LOGGER.error("Unknown error while logging into SimpliSafe: %s", err)
+            return self.async_show_form(
+                step_id="user",
+                data_schema=self.data_schema,
+                errors={"base": "unknown"},
+            )
+
+        return self.async_create_entry(
+            title=self._post_mfa_user_input[CONF_USERNAME],
+            data={
+                CONF_USERNAME: self._post_mfa_user_input[CONF_USERNAME],
+                CONF_TOKEN: simplisafe.refresh_token,
+                CONF_CODE: self._post_mfa_user_input.get(CONF_CODE),
+            },
+        )
 
     async def async_step_user(self, user_input=None):
         """Handle the start of the config flow."""
         if not user_input:
-            return await self._show_form()
+            return self.async_show_form(step_id="user", data_schema=self.data_schema)
 
         await self.async_set_unique_id(user_input[CONF_USERNAME])
         self._abort_if_unique_id_configured()
 
-        websession = aiohttp_client.async_get_clientsession(self.hass)
-
         try:
-            simplisafe = await API.login_via_credentials(
-                user_input[CONF_USERNAME], user_input[CONF_PASSWORD], session=websession
+            simplisafe = await self._async_get_simplisafe_api(user_input)
+        except PendingAuthorizationError:
+            LOGGER.info("Awaiting confirmation of MFA email click")
+            self._post_mfa_user_input = user_input
+            return self.async_show_form(step_id="mfa")
+        except InvalidCredentialsError:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=self.data_schema,
+                errors={"base": "invalid_credentials"},
+            )
+        except SimplipyError as err:
+            LOGGER.error("Unknown error while logging into SimpliSafe: %s", err)
+            return self.async_show_form(
```

Comment:
We can consolidate the show form calls by just setting the errors dict in the except blocks and below them check if `errors` is truthy and then show the form.